src/main/java/uk/gov/companieshouse/accounts/association/service/AssociationsService.java

```diff
@@ -191,4 +196,42 @@ public void updateAssociation(final String associationId, final Update update) {
 
     }
 
+    private AssociationDao filterForMostRecentInvitation( AssociationDao associationDao ){
+        final var invitations = associationDao.getInvitations();
+
+        if ( invitations.size() > 1 ){
+            final var mostRecentInvitation = invitations.stream()
+                    .max( Comparator.comparing( InvitationDao::getInvitedAt ) )
+                    .orElseThrow();
+
+            invitations.clear();
+            associationDao.setInvitations( List.of( mostRecentInvitation ) );
+        }
+
+        return associationDao;
+    }
+
+    @Transactional( readOnly = true )
+    public List<Invitation> fetchActiveInvitations( final String userId, final int pageIndex, final int itemsPerPage ){
+        return associationsRepository.fetchAssociationsWithActiveInvitations( userId, LocalDateTime.now() )
+                .map( this::filterForMostRecentInvitation )
+                .sorted( ( firstAssociation, secondAssociation ) -> {
```

Comment:
sorted, instead use

                .sorted(Comparator.comparing(AssociationDao::getApprovalExpiryAt))
                invited at will be exactly 7 days less than expiry at. Hence we can use expiry at instead.

Reply:
Addressed this, nice suggestion! https://github.com/companieshouse/accounts-association-api/pull/101/commits/33a3dffacf146595d164e418665b91b36eec3943